ols/src/prompts/prompt_generator.py

```diff
@@ -13,6 +13,11 @@
 from ols.customize import prompts
 
 
+def restructure_rag_context(text: str, model: str) -> str:
```

Comment:
this new method is effectively called in one place, why don't you drop this and simply call `resturcture_rag_context_pre(resturcture_rag_context_post(...))`? this would be more intuitive for reviewer.

Reply:
I think this is fine.  To me `restructure_rag_context_post(restructure_rag_context_pre(text, model), model)` looks more confusing.

Reply:
Yeah well it also was resulting easier to me to read this new function, shorter and more intuitive IMO, but anyway, up to you as well guys... thanks!